src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableValueBinder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableValueBinder<T> : IValueBinder
+    {
+        private DocumentDBContext _context;
+
+        public DocumentDBEnumerableValueBinder(DocumentDBContext context)
+        {
+            _context = context;
+        }
+
+        public Type Type
+        {
+            get
+            {
+                return typeof(IEnumerable<T>);
+            }
+        }
+
+        public object GetValue()
+        {
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(_context.ResolvedAttribute.DatabaseName, _context.ResolvedAttribute.CollectionName);
+
+            List<Document> finalResults = new List<Document>();
+
+            string continuation = null;
+
+            do
+            {
+                DocumentQueryResponse response = DocumentDBUtility.RetryAsync(
+                    () => _context.Service.ExecuteNextAsync(collectionUri, _context.ResolvedAttribute.SqlQuery, continuation),
```

Comment:
This value provider works for both queries as well as non query enumerables? E.g. if no SqlQuery is specified.